clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2511,4 +2511,39 @@ public void testEnforceRebalanceTriggersRebalanceOnNextPoll() {
 
         assertEquals(countingRebalanceListener.revokedCount, 1);
     }
+
+    @Test
+    public void deserializerShouldSeeGeneratedConfigs() {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, DeserializerForClientId.class.getName());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, DeserializerForClientId.class.getName());
+
+        KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(props);
+
+        assertEquals(2, DeserializerForClientId.CLIENT_IDS.size());
+        assertEquals(DeserializerForClientId.CLIENT_IDS.get(0), consumer.getClientId());
+        assertEquals(DeserializerForClientId.CLIENT_IDS.get(1), consumer.getClientId());
+
+        assertEquals(2, DeserializerForClientId.AUTO_COMMIT_FLAGS.size());
+        assertEquals(DeserializerForClientId.AUTO_COMMIT_FLAGS.get(0), consumer.isAutoCommitEnabled());
```

Comment:
This seems beyond the scope of this PR. Why do we also need to let deserializer see whether given consumer instance is auto commit enabled?

Reply:
This flag is included by reason of this comment https://github.com/apache/kafka/pull/9102#issuecomment-667734629

I’m ok to revert it if it is overkill